src/tscore/Throttler.cc

```diff
@@ -0,0 +1,42 @@
+/** @file
+
+  Implement Throttler.
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#include "tscore/Throttler.h"
+
+Throttler::Throttler(std::chrono::milliseconds periodicity) : _last_allowed_time{TimePoint{}}, _periodicity{periodicity} {}
```

Comment:
Still unclear on what the member initializer for `_last_allowed_time` is doing that is useful.

Reply:
The explicit initialization seems to be required for some reason. See:
https://github.com/bneradt/trafficserver/pull/1#discussion_r505812023

But I can move it to the header. Is that better?

Reply:
Yeah, I agree that is strange.

I seem to only have to do this with gcc 8 and 9. gcc 10 is fine without the explicit default construction as we'd expect:
https://stackoverflow.com/q/64393225/629530

Talking offline, we agreed that default initializing this in the header is at least a cleaner solution than here. I moved it there.

Thanks for inquiring into this.